src/main/java/io/cdap/plugin/aws/sqs/source/SqsSourceUtil.java

```diff
@@ -0,0 +1,210 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.aws.sqs.source;
+
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.model.Message;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.amazonaws.services.sqs.model.SetQueueAttributesRequest;
+import io.cdap.cdap.api.data.format.FormatSpecification;
+import io.cdap.cdap.api.data.format.RecordFormat;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.streaming.StreamingContext;
+import io.cdap.cdap.format.RecordFormats;
+import io.cdap.plugin.aws.sqs.exception.SqsInitializationException;
+import io.cdap.plugin.aws.sqs.util.SqsUtil;
+import org.apache.spark.storage.StorageLevel;
+import org.apache.spark.streaming.api.java.JavaDStream;
+import org.apache.spark.streaming.api.java.JavaReceiverInputDStream;
+import org.apache.spark.streaming.receiver.Receiver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Util method for {@link SqsSource}.
+ *
+ * <p>This class contains methods for {@link SqsSource} that require spark classes because
+ * during validation spark classes are not available. Refer CDAP-15912 for more information.
+ */
+final class SqsSourceUtil {
+  private static final Logger LOG = LoggerFactory.getLogger(SqsSourceUtil.class);
+
+  private SqsSourceUtil() {
+    // no-op
+  }
+
+  static JavaDStream<StructuredRecord> getJavaDStream(StreamingContext context, SqsSourceConfig conf) {
+    return context.getSparkStreamingContext().receiverStream(SqsSourceUtil.getReceiver(conf));
+  }
+
+  /**
+   * Gets {@link JavaReceiverInputDStream} for {@link SqsSource}.
+   *
+   * @param conf {@link SqsSourceConfig} config
+   */
+  private static Receiver<StructuredRecord> getReceiver(SqsSourceConfig conf) {
+    return new Receiver<StructuredRecord>(StorageLevel.MEMORY_ONLY()) {
+      @Override
+      public StorageLevel storageLevel() {
+        return StorageLevel.MEMORY_ONLY();
+      }
+
+      @Override
+      public void onStart() {
+        new Thread() {
+          @Override
+          public void run() {
+            AmazonSQS sqs = null;
+            try {
+              sqs = SqsUtil.getSqsClient(conf.getAuthenticationMethod(), conf.getAccessId(),
```

Comment:
Do we have to shutdown this client on stop?

Reply:
Done